pkg/postgresql/postgresql.go

```diff
@@ -428,19 +428,22 @@ func (p *Manager) WriteRecoveryConf(followedConnParams ConnParams) error {
 }
 
 func (p *Manager) writePgHba() error {
-	f, err := os.OpenFile(filepath.Join(p.dataDir, "pg_hba.conf"), os.O_APPEND|os.O_RDWR, 0)
-	if err != nil {
-		return err
+	var fileContent []byte
+	var content = [][]byte{
+		[]byte("local all all  trust\n"),
+		[]byte("host all all 127.0.0.1/32 trust\n"),
+		[]byte("host all all ::1/128 trust\n"),
+		[]byte("hostssl all all 0.0.0.0/0 md5\n"),
+		[]byte("hostssl all all ::0/0 md5\n"),
+		[]byte(fmt.Sprintf("hostssl replication %s %s md5\n", p.replUsername, "0.0.0.0/0")),
```

Comment:
Why not 
```go
fmt.Sprintf("hostssl replication %s 0.0.0.0/0 md5\n", p.replUsername, )
```
e.g. avoiding the format with a constant by embedding it directly?

Apart from that, you could work with `[]string` to avoid many  type-casts:
```go
content := []string{
		"local all all  trust",
		"host all all 127.0.0.1/32 trust",
		...
}
```
and then cast at the call point:
```go
return common.WriteFileAtomic(filepath.Join(p.dataDir, "pg_hba.conf"), []byte(strings.Join(contents, "\n")), 0600)
```